### DISCUSSION TITLE
Any plans to move away from xdg-desktop-portal-gnome?

### DISCUSSION THREAD

**Discussion by coatlessali** [niri-wm/niri#763] [General]
Niri has been great in the few months I have been using it, but I do have one thing I don't really like: `xdg-desktop-portal-gnome` pulls `gnome-desktop` as a dependency, and I'd like to get away from it if possible. Are there any plans to implement niri's own portal, or support other portals?

**Comment by YaLTeR** [comment]
There are no plans because, well, I'm a single person with limited free time, and a portal would be a whole separate project to maintain. Though, another argument for a niri portal is to expose more unusual screencast options, like workspace screencasting.

> pulls gnome-desktop as a dependency

Why is this a problem? It's a small utility library: https://gitlab.gnome.org/GNOME/gnome-desktop/

**Comment by Piroro-hs** [comment]
I think that the real problem of xdp-gnome is its libx11 dependency. (Sorry for necrobumping)

**Reply by YaLTeR** [reply]
xdp-gtk probably also has that, and that's the fallback impl that will always be used for some portals.

**Comment by nebulosa2007** [comment]
Btw, on ArchLinux you could choose which portal you prefer to use: **xdg-desktop-portal-impl** ([xdg-desktop-portal-gtk](https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal-gtk/), [xdg-desktop-portal-kde](https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal-kde/), [xdg-desktop-portal-gnome](https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal-gnome/), [xdg-desktop-portal-hyprland](https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal-hyprland/), [xdg-desktop-portal-lxqt](https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal-lxqt/), [xdg-desktop-portal-wlr](https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal-wlr/), [xdg-desktop-portal-xapp](https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal-xapp/), [xdg-desktop-portal-dde](https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal-dde/), [xdg-desktop-portal-cosmic](https://archlinux.org/packages/extra/x86_64/xdg-desktop-portal-cosmic/), [xdg-desktop-portal-kde](https://archlinux.org/packages/kde-unstable/x86_64/xdg-desktop-portal-kde/), [xdg-desktop-portal-gnome](https://archlinux.org/packages/gnome-unstable/x86_64/xdg-desktop-portal-gnome/))

[Niri package](https://archlinux.org/packages/extra/x86_64/niri/)

So author should ask a maintainer for dependencies update

**Reply by YaLTeR** [reply]
niri depends on xdp-gnome for screencasting. No other portals will work (except maybe -wlr).

**Reply by coatlessali** [reply]
Decided to test all of the ones that support screencasting. Here are the results:

- xdg-desktop-portal-wlr results in a flickery, unusable mess that updates every 3 seconds or so and rapidly flickers between two frames.
- xdg-desktop-portal-hyprland results in smooth video, but repeating previous frames and flickering as a result.
- Everything else doesn't work at all. No selection menu appears, properties on pipewire screencasting in OBS are inaccessible.

My actual issue isn't necessarily `gnome-desktop` as I had originally thought - didn't bother to check and that's my bad. The actual issue I have is that `xdg-desktop-portal-gnome` is dependent on `nautilus`.  It's really not actually a big deal though. Going to see if I can change the default file picker by editing the portals file.

**Reply by YaLTeR** [reply]
Removing FileChooser from `/usr/share/xdg-desktop-portal/portals/gnome.portal` will probably make it fall back to the xdp-gtk built-in file chooser, without nautilus.

**Reply by Yaroslav-95** [reply]
I really dislike the nautilus/gnome file chooser for several reasons, among them being the fact that it starts some localsearch process that I don't need because I don't use nautilus, it's very sluggish and constantly crashes on me. So I solved it for me by adding the following line to `/usr/share/xdg-desktop-portal/niri-portals.conf`:

```
org.freedesktop.impl.portal.FileChooser=gtk;
```

Now apps in my setup open what is, IMHO, a way more sane file chooser that just works™.

**Reply by coatlessali** [reply]
I instead copied it to `/usr/local/` and then edited it. Works just fine.

**Reply by Yaroslav-95** [reply]
That also worked for me. Thanks for the tip!

**Reply by ilylily** [reply]
there is also the xdg config dir (default `$HOME/.config`), per the xdg portal docs https://flatpak.github.io/xdg-desktop-portal/docs/portals.conf.html

**Comment by aacebedo** [comment]
a note about xdg-desktop-portal-wlr, I tested the latest version that is now supporting individual window sharing. It seems to works for the entire output but I can't see individual windows in the chooser. 
I guess something is missing on Niri side but I was wondering what exactly? 
I am not a wayland specialist but it seems to be linked to the ext_foreign_toplevel_list_v1 interface.

There is an issue discussing the subject here and in which the implementation has been acknowledged: 
https://github.com/emersion/xdg-desktop-portal-wlr/issues/107

**Reply by YaLTeR** [reply]
We don't have ext- screencopy yet

**Reply by aacebedo** [reply]
will it be on the roadmap?

**Reply by YaLTeR** [reply]
there is no roadmap

**Reply by itshog** [reply]
How difficult would it be for Niri to support individual window sharing through the wlr portal? Despite what some older comments say screen sharing with xdg-desktop-portal-wlr works great as of now, being able to share individual windows without pulling in nautilus and gvfs would be a very nice addition

**Comment by skyhighcaterpie** [comment]
CachyOS Niri is not out-of-box like KDE Plasma.
Nautilus is default file manager, but it is a bit lag on my Nvidia 4060 Laptop.

I am just a normal user from Windows, not a engineer, have no knowledge on Linux or writing script. 
There is no official guide for using KDE Plasma software on CachyOS Niri. 
I try copy & paste environment variables code from reddit, but it results in a endless pop-up and I have to reinstall CachyOS.

After several month trial and error with google, I finally figure out how to make Dolphin works on CachyOS Niri:
sudo pacman -S archlinux-xdg-menu
sudo ln -s /etc/xdg/menus/plasma-applications.menu /etc/xdg/menus/applications.menu
XDG_MENU_PREFIX=plasma- kbuildsycoca6 --noincremental

Niri scrolling is really cool, but mixing Niri with KDE Plasma software is a tough one for normal people.

**Reply by stefonarch** [reply]
> Niri scrolling is really cool, but mixing Niri with KDE Plasma software is a tough one for normal people.

That's true as both Plasma and Gnome don't really care about other DE/WMs IMO.  What should work out with Qt of the box is the  LQXt session with niri as compositor. File manager opening for example is instant as it draws also the desktop and is running as a daemon; menus and file associations are no issue too.
